olive/passes/onnx/transformer_optimization.py

```diff
@@ -94,30 +94,62 @@ def sd_model_types():
-        if not is_ort_115_or_newer and input_model_type != "unet":
+        input_model_type = run_config["model_type"]
+        if not is_ort_1_15_0_or_newer and input_model_type != "unet":
             # 'vae' and 'clip' are only recognized in ORT v1.15+. earlier versions of ORT stable diffusion
             # optimization simply use "unet" for these model types.
             run_config["model_type"] = "unet"
 
+        from onnxruntime.transformers.fusion_options import FusionOptions
+
         fusion_options = FusionOptions(run_config["model_type"])
 
-        if input_model_type == "unet":
-            fusion_options.enable_packed_kv = use_float16
-            fusion_options.enable_packed_qkv = use_float16 and is_ort_115_or_newer
-            fusion_options.enable_bias_add = is_ort_115_or_newer
+        # TODO: remove dml_future when ORT 1.15.1 with future version of DML is released
+        # This "provider" value is simply a way to test in-development changes without having
+        # to bump the ORT version.
+        dml_future = pass_config["target_provider"] == "directml_future"
+
+        if pass_config["target_provider"] == "directml" or dml_future:
+            # Some of these fusions are disabled because they provide no performance advantage,
+            # and it's preferable to limit ops outside the ONNX domain.
+            fusion_options.enable_gelu = is_ort_1_15_0_or_newer
+            fusion_options.enable_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_attention = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.use_multi_head_attention = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.enable_skip_layer_norm = False
+            fusion_options.enable_embed_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_bias_skip_layer_norm = False
+            fusion_options.enable_bias_gelu = is_ort_1_15_0_or_newer
+            fusion_options.enable_gelu_approximation = False
+            fusion_options.enable_qordered_matmul = False
+            fusion_options.enable_shape_inference = is_ort_1_15_0_or_newer
+            fusion_options.enable_gemm_fast_gelu = False
+            fusion_options.enable_nhwc_conv = False
+            fusion_options.enable_group_norm = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.enable_bias_splitgelu = False
+            fusion_options.enable_packed_qkv = pass_config["float16"] and is_ort_1_15_0_or_newer
+            fusion_options.enable_packed_kv = pass_config["float16"] and is_ort_1_15_0_or_newer
+            fusion_options.enable_bias_add = False
+        else:
+            if input_model_type == "unet":
+                fusion_options.enable_packed_kv = pass_config["float16"]
+                fusion_options.enable_packed_qkv = pass_config["float16"] and is_ort_1_15_0_or_newer
+                fusion_options.enable_bias_add = is_ort_1_15_0_or_newer
```

Comment:
`enable_bias_add` is False above, but true here. 🤔 Confirming that's intended? Do you know if that emits DML ADD1?

Reply:
This is to mirror logic in upstream ORT stable diffusion scripts for non-DML EP.